src/local/specs/local-auth.guard.spec.ts

```diff
@@ -0,0 +1,125 @@
+import { LocalAuthGuard } from '../local-auth.guard'
+import { ExecutionContext, UnauthorizedException } from '@nestjs/common'
+import { createMock } from '@golevelup/ts-jest'
+import { LocalAuthStrategy } from '../local-auth.strategy'
+
+const injectStrategy = () =>
+  new LocalAuthStrategy({
+    handleLoginFor: (username, password) =>
+      Promise.resolve({ username, password, id: '1' }),
+  })
+
+const injectCustomizedUsernameStrategy = () =>
+  new LocalAuthStrategy({
+    strategyOptions: {
+      usernameField: 'mail',
+    },
+    handleLoginFor: (username, password) =>
+      Promise.resolve({ username, password, id: '1' }),
+  })
+
+const injectCustomizedPasswordStrategy = () =>
+  new LocalAuthStrategy({
+    strategyOptions: {
+      passwordField: 'pass',
+    },
+    handleLoginFor: (username, password) =>
+      Promise.resolve({ username, password, id: '1' }),
+  })
+
+describe('LocalAuthGuard', () => {
+  let localAuthGuard: LocalAuthGuard
+
+  beforeEach(() => {
+    localAuthGuard = new LocalAuthGuard()
+  })
+
+  it('should throw an error when there is no related strategy', async () => {
+    const context = createMock<ExecutionContext>()
+    await expect(localAuthGuard.canActivate(context)).rejects.toThrow(
+      'Unknown authentication strategy "local"',
+    )
+  })
+
+  it('should throw and Unauthorized error when username field and password field are undefined', async () => {
```

Comment:
You have a typo here. Replace and for an. Also, we should stick to a format. Maybe the one we've already use for JwtAuthGuard